packages/flame_rive/lib/src/rive_component.dart

```diff
@@ -0,0 +1,177 @@
+import 'dart:async';
+import 'dart:math';
+import 'dart:ui' as ui;
+
+import 'package:flame/components.dart';
+import 'package:flutter/rendering.dart';
+import 'package:rive/rive.dart';
+
+// ignore_for_file: implementation_imports
+import 'package:rive/src/rive_core/math/aabb.dart';
+import 'package:rive/src/rive_core/math/mat2d.dart';
+import 'package:rive/src/rive_core/math/vec2d.dart';
+
+class RiveComponent extends PositionComponent {
+  final Artboard artboard;
+  final RiveArtboardRenderer _renderer;
+
+  RiveComponent({
+    required this.artboard,
+    bool antialiasing = true,
+    bool useArtboardSize = true,
+    BoxFit fit = BoxFit.contain,
+    Alignment alignment = Alignment.center,
+
+    // position component arguments
+    Vector2? position,
+    Vector2? size,
+    Vector2? scale,
+    double angle = 0.0,
+    Anchor anchor = Anchor.topLeft,
+    int? priority,
+  })  : _renderer = RiveArtboardRenderer(
+          antialiasing: antialiasing,
+          useArtboardSize: useArtboardSize,
+          fit: fit,
+          alignment: alignment,
+          artboard: artboard,
+        ),
+        super(
+          position: position,
+          size: size,
+          scale: scale,
+          angle: angle,
+          anchor: anchor,
+          priority: priority,
+        );
+
+  @override
+  void render(ui.Canvas canvas) {
+    super.render(canvas);
+    _renderer.render(canvas, size.toSize());
+  }
+
+  @override
+  void update(double dt) {
+    super.update(dt);
+    _renderer.advance(dt);
+  }
+}
+
+class RiveArtboardRenderer {
```

Comment:
Maybe this should be a separate file?

Reply:
Totally disagree since it is an integral part of the `RiveComponent`.

Reply:
Should it be a private class even then maybe?

Reply:
Also:
https://dart.dev/guides/language/effective-dart/design#consider-declaring-multiple-classes-in-the-same-library